testserv.c

```diff
@@ -141,30 +165,39 @@ kill_srvpid(void)
     srvpid = 0;
 }
 
-#define SERVER() (progname=__func__, mustforksrv())
+#define SERVER() (progname=__func__, mustforksrv(NULL))
 
 // Forks the server storing the pid in srvpid.
 // The parent process returns port assigned.
 // The child process serves until the SIGTERM is received by it.
 static int
-mustforksrv(void)
+mustforksrv(char *socket_file)
```

Comment:
I would prefer not to mix UNIX socket into this function. This function returns an int, port value. Your change to this function looks confusing. I suggest to copy this function to mustforksrv_unix(). This function should return a socket path. And define macro SERVER_UNIX() similar to SERVER().
It is okay to have dup code in those functions for now. Simplicity is more important than elegance or cleverness.

Reply:
Sure.  But I'm blocked by #541.  So I won't be able to update this PR until #541 is merged.

Reply:
Ok, let's wait for it. I won't plan any other merges too to not delay it further...

Reply:
I have merged #541.